pages/blog.js

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import styled from 'styled-components';
+import Head from 'next/head';
+import { useRouter } from 'next/router';
+import Link from 'next/link';
+import PropTypes from 'prop-types';
+import Layout from '../components/Layout';
+import Primary from '../components/Primary';
+import BlogFeed from '../components/BlogFeed';
+
+export async function getStaticProps() {
+  const posts = (context => {
+    const keys = context.keys();
+    const values = keys.map(context);
+
+    const data = keys.map((key, index) => {
+      const slug = key.replace(/^.*[\\/]/, '').slice(0, -3);
+      const value = values[index];
+      return {
+        attributes: value.attributes,
+        markdownBody: value.html,
+        slug,
+      };
+    });
+    return data;
+  })(require.context('../content/blog/posts', true, /\.md$/));
+
+  return {
+    props: {
+      posts,
+    },
+  };
+}
+
+const Blog = ({ posts, ...props }) => {
+  const perPage = 5;
+  const pageCount = Math.ceil(posts.length / perPage);
+
+  const router = useRouter();
+  let { page } = router.query;
+  page = parseInt(page);
+  if (Number.isNaN(page) || page < 1 || page > pageCount) {
+    page = 1;
+  }
+
+  const pageLinks = [];
+  for (let pageNumber = 1; pageNumber <= pageCount; pageNumber += 1) {
+    if (pageNumber === page) {
+      pageLinks.push(
+        <PageNumber className="current" key={pageNumber}>
+          {pageNumber}
+        </PageNumber>
+      );
+    } else {
+      const pagePath = `/blog?page=${pageNumber}`;
+
+      pageLinks.push(
+        <Link href={pagePath} key={pageNumber}>
+          <a href={pagePath} style={{ textDecoration: 'none' }}>
+            <PageNumber>{pageNumber}</PageNumber>
```

Comment:
What do you think about holding off on pagination until we hit 10 or so blog posts? Feels a little strange to have a "1" at the bottom of the page with only one page.